### PR TITLE
doc: Convert README from ReST to Markdown

### LINKED ISSUE #2928: Convert documentation source files from ReStructuredText to Markdown
Currently, some documentations are written in Markdown while some are in ReStructuredText. I think it's better to use a consistent format so that new contributors won't be confused.

Here are the `.rst` files in the repository:
```
$ find . -name '*.rst'
./README.rst
./doc/install.rst
./doc/overview.rst
./doc/index.rst
./doc/api/index.rst
./doc/_templates/autosummary/module.rst
./doc/_templates/autosummary/class.rst
./doc/_templates/autosummary/exception.rst
./doc/_templates/autosummary/method.rst
./doc/_templates/autosummary/base.rst
./doc/_templates/autosummary/function.rst
```
I think at least the following three files should be converted to markdown:

- [ ] `README.rst` #3059 
- [x] `doc/install.rst` #2992 
- [x] `doc/overview.rst` #2953

Comments?

### PR DESCRIPTION
**Pull request by yvonnefroehlich**
**Description of proposed changes**

This PR converts the REAMDE from ReStructuredText to Markdown.
- [x] Use `pandoc` to convert to markdown
- [x] Make manual adjustments
- [x] Fix image
- [x] Update files
- [x] Fix appearance on webpage
       Adding `.. doc-index-start-after` and `.. doc-index-end-before` does not work correctly. We have to look for the corresponding Markdown syntax

**Preview**:
- GitHub
  - ReST (old): https://github.com/GenericMappingTools/pygmt/blob/main/README.rst
  - Markdown (new): https://github.com/GenericMappingTools/pygmt/tree/convert-readme-rst2md?tab=readme-ov-file#pygmt
- Website
  - ReST (old): https://www.pygmt.org/dev/index.html
  - Markdown (new): https://pygmt-dev--3059.org.readthedocs.build/en/3059/

Fixes #2928

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash command is:

- `/format`: automatically format and lint the code

### CODE REVIEW

**Comment by seisman** [comment]
Need to update line 12 in `doc/index.rst`.

File: README.md [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+    data tables, `xarray.DataArray` for grids and `geopandas.GeoDataFrame` for geographical data.
+
+## Quickstart
+
+### Installation
+
+Simple installation using [mamba](https://mamba.readthedocs.org/):
+
+```bash
+    mamba install --channel conda-forge pygmt
```
Comment (seisman) [inline README.md:57]:
```suggestion
mamba install --channel conda-forge pygmt
```

File: README.md [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+```
+
+If you use [conda](https://docs.conda.io/projects/conda/en/latest/user-guide/index.html):
+
+```bash
+    conda install --channel conda-forge pygmt
```
Comment (seisman) [inline README.md:63]:
```suggestion
conda install --channel conda-forge pygmt
```

File: README.md [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+### Getting started
+
+As a starting point, you can open a [Python interpreter](https://docs.python.org/3/tutorial/interpreter.html)
+or a [Jupyter notebook](https://docs.jupyter.org/en/latest/running.html), and try the following example:
+
+``` python
+    import pygmt
+    fig = pygmt.Figure()
+    fig.coast(projection="H10c", region="g", frame=True, land="gray")
+    fig.show()
```
Comment (seisman) [inline README.md:77]:
```suggestion
import pygmt
fig = pygmt.Figure()
fig.coast(projection="H10c", region="g", frame=True, land="gray")
fig.show()
```

File: doc/maintenance.md [hunk 4]
```diff
@@ -231,7 +231,7 @@ publishing the actual release notes at {doc}`changes`.
 
 ### Check the README Syntax
 
-GitHub is a bit forgiving when it comes to the RST syntax in the README but PyPI is not.
+GitHub is a bit forgiving when it comes to the syntax in the README but PyPI is not.
 To check the README syntax, visit the
 [PyGMT TestPyPI release history](https://test.pypi.org/project/pygmt/#history), select
 the latest commit, and review the left sidebar and project description for any errors.
```
Comment (yvonnefroehlich) [inline doc/maintenance.md:237]: Do we want to keep this subsection in the Maintainers Guide as a reminder (for new maintainers / contributors), or do people think we can remove it completely?

**Comment by yvonnefroehlich** [comment]
> `README.rst` is hard-coded in many files. Need to update `README.rst` to `README.md` in all these files:
> 
> ```
> pyproject.toml:8:readme = "README.rst"
> doc/maintenance.md:225:9. Refresh citation information. Specifically, the BibTeX in `README.rst` and
> doc/index.rst:13:.. include:: ../README.rst
> .github/workflows/ci_docs.yml:26:      - 'README.rst'
> .github/workflows/ci_docs.yml:35:      - 'README.rst'
> .github/workflows/publish-to-pypi.yml:27:      - 'README.rst'
> .readthedocs.yaml:24:              README.rst \
> ```

Thanks for posting this list! These files should be updated in the commits  https://github.com/GenericMappingTools/pygmt/pull/3059/commits/3360f455002f4b7023b8a0fa4461f6d60f1ebfbe, https://github.com/GenericMappingTools/pygmt/pull/3059/commits/7be24c313086bcd369535e0b1005aedc7ebe7074, and https://github.com/GenericMappingTools/pygmt/pull/3059/commits/a68367c3697e9a8361407dfa06e86854354e381b.

**Comment by yvonnefroehlich** [comment]
Currently building the documentation fails and the issue seem to be related to downloading the GMT remote datasets:
```
    pygmt.exceptions.GMTCLibError: Module 'grdcut' failed with status code 72:
    grdcut [ERROR]: Remote download is currently deactivated
    grdcut [ERROR]: Unable to obtain remote file @earth_relief_10m_g
```
Is there currently any maintenance regarding the remote datasets and they are temporarily not available?
This issue is also reported in #3063.

**Comment by MarcoSal** [comment]
> Currently building the documentation fails and the issue seem to be related to downloading the GMT remote datasets:
> 
> ```
>     pygmt.exceptions.GMTCLibError: Module 'grdcut' failed with status code 72:
>     grdcut [ERROR]: Remote download is currently deactivated
>     grdcut [ERROR]: Unable to obtain remote file @earth_relief_10m_g
> ```
> 
> Is there currently any maintenance regarding the remote datasets and they are temporarily not available? This issue is also reported in #3063.

now it's working again. the service was probably under maintenance.

**Comment by yvonnefroehlich** [comment]
It looks like downloading the remote datasets works again and this was only a temporary issue.

**Comment by seisman** [comment]
> * Fix appearance on webpage
>   Adding `.. doc-index-start-after` and `.. doc-index-end-before` does not work correctly. We have to look for the corresponding Markdown syntax

Does the solution here 
https://stackoverflow.com/questions/46278683/include-my-markdown-readme-into-sphinx/68005314#68005314 work for you?

File: doc/maintenance.md (hunk 4 above)
Comment (seisman) [reply doc/maintenance.md:237]: I think we should still keep it, since there are also many Markdown flavors.

File: doc/index.rst [hunk 5]
```diff
@@ -10,7 +10,7 @@
         </h2>
     </div>
 
-.. include:: ../README.rst
+.. include:: ../README.md
     :start-after: doc-index-start-after
     :end-before: doc-index-end-before
```
Comment (seisman) [inline doc/index.rst:15]:
~~~suggestion
```{include} ../README.md
:start-after: doc-index-start-after
:end-before: doc-index-end-before
```
~~~

File: doc/maintenance.md (hunk 4 above)
Comment (yvonnefroehlich) [reply doc/maintenance.md:237]: I am fine with keeping this subsection in the Maintainers Guide!

File: doc/index.rst (hunk 5 above)
Comment (yvonnefroehlich) [reply doc/index.rst]: I tried this idea, but unfortunately, this also does not work:

| GitHub | Website |
| --- | --- |
| include_md_in_rst_GitHub | include_md_in_rst_Webside |

Comment (seisman) [reply doc/index.rst]: perhaps you have to convert index.rst to index.md

Comment (yvonnefroehlich) [reply doc/index.rst]: I will try this, but I will be quite busy in the next three weeks. Thus, I am not sure if I will have the time to do this in the next days 🙁.

File: doc/maintenance.md (hunk 4 above)
Comment (seisman) [inline doc/maintenance.md:237]: https://packaging.python.org/en/latest/guides/making-a-pypi-friendly-readme/#creating-a-readme-file

PyPI uses the [GitHub Flavored Markdown](https://github.github.com/gfm/) syntax by default, so we no longer need to check the README syntax on PyPI.

Comment (seisman) [reply doc/maintenance.md]: Done in 13426785a1985df61496ec1286cf80b2da3f9298.

**Review by seisman — APPROVED** [review]

**Review by michaelgrund — APPROVED** [review]

File: doc/index.rst (hunk 5 above)
Comment (yvonnefroehlich) [reply doc/index.rst]: Thanks @seisman for doing this and completing this PR.

**Pull request merged**